app/setup.py

```diff
@@ -228,36 +227,26 @@ def after_request(response):  # pylint: disable=unused-variable
     return application
 
 
-def _get_cdn_root_url(application) -> str:
-    parsed_url = urlparse(application.config["CDN_URL"])
-    return f"{parsed_url.scheme}://{parsed_url.hostname}"
-
-
-def add_cdn_url_to_csp_policies(application) -> Dict:
+def add_cdn_url_to_csp_policies(cdn_url) -> Dict:
     csp_policy = deepcopy(CSP_POLICY)
-    cdn_root_url = _get_cdn_root_url(application)
-    updated_policy = {}
-    for k, v in csp_policy.items():
-        updated_policy[k] = v
-
-        if k != "frame-src":
-            updated_policy[k].append(cdn_root_url)
-
-    return updated_policy
+    for policy in csp_policy:
+        if policy != "frame-src":
+            csp_policy[policy].append(cdn_url)
```

Comment:
This [google doc](https://developers.google.com/web/fundamentals/security/csp#policy_applies_to_a_wide_variety_of_resources) refers to the elements of the policy as directives.
```suggestion
    for directive in csp_policy:
        if directive != "frame-src":
            csp_policy[directive].append(cdn_url)
```